local_planner/test/test_planner_functions.cpp

```diff
@@ -0,0 +1,560 @@
+#include <gtest/gtest.h>
+
+#include <cmath>
+
+#include "../src/nodes/planner_functions.h"
+
+using namespace avoidance;
+
+TEST(PlannerFunctionsTests, calculateFOV) {
+  // GIVEN: the horizontal and vertical Field of View, the vehicle yaw and pitc
+  double h_fov = 90.0;
+  double v_fov = 45.0;
+  double yaw_z_greater_grid_length =
+      3.14;  // z_FOV_max >= GRID_LENGTH_Z && z_FOV_min >= GRID_LENGTH_Z
+  double yaw_z_max_greater_grid =
+      -2.3;  // z_FOV_max >= GRID_LENGTH_Z && z_FOV_min < GRID_LENGTH_Z
+  double yaw_z_min_smaller_zero = 3.9;  // z_FOV_min < 0 && z_FOV_max >= 0
+  double yaw_z_smaller_zero = 5.6;      // z_FOV_max < 0 && z_FOV_min < 0
+  double pitch = 0.0;
+
+  // WHEN: we calculate the Field of View
+  std::vector<int> z_FOV_idx_z_greater_grid_length;
+  std::vector<int> z_FOV_idx_z_max_greater_grid;
+  std::vector<int> z_FOV_idx3_z_min_smaller_zero;
+  std::vector<int> z_FOV_idx_z_smaller_zero;
+  int e_FOV_min;
+  int e_FOV_max;
+
+  calculateFOV(h_fov, v_fov, z_FOV_idx_z_greater_grid_length, e_FOV_min,
+               e_FOV_max, yaw_z_greater_grid_length, pitch);
+  calculateFOV(h_fov, v_fov, z_FOV_idx_z_max_greater_grid, e_FOV_min, e_FOV_max,
+               yaw_z_max_greater_grid, pitch);
+  calculateFOV(h_fov, v_fov, z_FOV_idx3_z_min_smaller_zero, e_FOV_min,
+               e_FOV_max, yaw_z_min_smaller_zero, pitch);
+  calculateFOV(h_fov, v_fov, z_FOV_idx_z_smaller_zero, e_FOV_min, e_FOV_max,
+               yaw_z_smaller_zero, pitch);
+
+  // THEN:
```

Comment:
Could you give a little explanation for what the checks do here? What are the different cases?